### PR TITLE
improve exceptions when parsing malformed files

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15702 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Routing/Loader/YamlFileLoader.php [hunk 1]
```diff
@@ -60,7 +61,11 @@ public function load($file, $type = null)
             $this->yamlParser = new YamlParser();
         }
 
-        $config = $this->yamlParser->parse(file_get_contents($path));
+        try {
+            $config = $this->yamlParser->parse(file_get_contents($path));
+        } catch (ParseException $e) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not contain valid YAML: %s', $path, $e->getMessage()));
```
Comment (Koc) [inline src/Symfony/Component/Routing/Loader/YamlFileLoader.php]: You could pass `$e` as third argument of `InvalidArgumentException` constructor

Comment (xabbuh) [inline src/Symfony/Component/Routing/Loader/YamlFileLoader.php]: good idea

**Comment by dupuchba** [comment]
Don't you need to add tests in order to check if the exception is thrown ?

**Comment by fabpot** [comment]
:+1: 
Status: Reviewed

**Comment by Tobion** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @xabbuh.

File: src/Symfony/Component/Validator/Mapping/Loader/YamlFileLoader.php [hunk 2]
```diff
@@ -117,7 +118,11 @@ protected function parseNodes(array $nodes)
      */
     private function parseFile($path)
     {
-        $classes = $this->yamlParser->parse(file_get_contents($path));
+        try {
+            $classes = $this->yamlParser->parse(file_get_contents($path));
+        } catch (ParseException $e) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not contain valid.', $path), 0, $e);
```
Comment (DavidBadura) [inline src/Symfony/Component/Validator/Mapping/Loader/YamlFileLoader.php]: missing "YAML" in exception message. fixed in #15749

File: src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php [hunk 3]
```diff
@@ -286,7 +287,13 @@ protected function loadFile($file)
             $this->yamlParser = new YamlParser();
         }
 
-        return $this->validate($this->yamlParser->parse(file_get_contents($file)), $file);
+        try {
+            $configuration = $this->yamlParser->parse(file_get_contents($file));
+        } catch (ParseException $e) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not contain valid YAML.', $file), 0, $e);
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php]: this should probably use the exception of the component instead (as done when the file does not exist)

Comment (jakzal) [inline src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php]: fixed in #15750

**Pull request merged**